src/events/stickyMessageHandler.spec.ts

```diff
@@ -0,0 +1,115 @@
+import { Message } from 'discord.js'
+
+import { StickyMessage } from '@prisma/client'
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
+
+import * as channelLock from '../features/stickyMessage/channelLock.js'
+import * as stickyMessage from '../features/stickyMessage/index.js'
+import * as messageCounter from '../features/stickyMessage/messageCounter.js'
+import { BotContext } from '../types/BotContext.js'
+import * as cache from '../utils/cache.js'
+
+import stickyMessageEventHandler from './stickyMessageHandler.js'
+
+vi.mock('../config.js', async () => {
+  const Environment = {
+    MESSAGE_COOLDOWN_SEC: 15,
+    MESSAGE_MAX: 5,
+  }
+
+  return { Environment }
+})
+
+describe('stickyMessageEventHandler', () => {
+  const channelId = 'test-channel'
+  let message: Message
+  let stickyMessageEntity: StickyMessage
+
+  beforeEach(() => {
+    message = { channelId } as unknown as Message
+    stickyMessageEntity = {} as unknown as StickyMessage
```

Comment:
I don't thing as unknown is necessary for casting

Reply:
do you have any suggestion to mock discordJS object.
I try to initialize the discordJS class such as `message` or `channel`, but unfortunately it is the protected constructor that can only initialize by the Discord class.I solved the mocking problem by using unknown like this :<

Reply:
For this comment I mean you don't have to cast `as unknown` first.
You can directly cast to `Message` for example.
```ts
message = { channelId } as Message
```
I've test this and it's fine.

but if you want to use private constructor, I don't know if others will allow this syntax 😅
```ts
message = new (Message as any)()
```

Reply:
ohh, I see